apps/rush-lib/src/logic/InstallManager.ts

```diff
@@ -1029,6 +1064,49 @@ export class InstallManager {
     });
   }
 
+  private _mergeEnvironmentVariables(
+    baseEnv: NodeJS.ProcessEnv,
+    environmentVariables?: IConfigurationEnvironment
+  ): NodeJS.ProcessEnv {
+    const packageManagerEnv: NodeJS.ProcessEnv = baseEnv;
+
+    if (environmentVariables) {
+      // eslint-disable-next-line guard-for-in
+      for (const envVar in environmentVariables) {
+        let setEnvironmentVariable: boolean = true;
+        console.log(`\nProcessing definition for environment variable: ${envVar}`);
+
+        if (baseEnv.hasOwnProperty(envVar)) {
+          setEnvironmentVariable = false;
+          console.log(colors.yellow(`WARNING: Environment variable already defined:`));
```

Comment:
This shouldn't be a warning if they explicitly said they want to override it.

```suggestion
          console.log(`WARNING: Environment variable already defined:`);
```